kotlin/workflow-runtime/src/main/java/com/squareup/workflow/debugging/WorkflowHierarchyDebugSnapshot.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow.debugging
+
+import com.squareup.workflow.debugging.WorkflowHierarchyDebugSnapshot.Child
+import com.squareup.workflow.internal.WorkflowId
+
+/**
+ * Describes a tree of snapshots of the states of the entire workflow tree after a particular render
+ * pass.
+ *
+ * Emitted from [com.squareup.workflow.WorkflowSession.debugSnapshots].
+ *
+ * @param workflowType A string representation of the type of this workflow.
+ * @param stateDescription A (lazy) string representation of the state of this workflow after the
+ * last render pass.
+ * @param children All the child workflows that were rendered by this workflow in the last render
+ * pass. See [Child].
+ */
+data class WorkflowHierarchyDebugSnapshot(
+  val workflowType: String,
+  val stateDescription: LazyString,
```

Comment:
I was too squeamish about the performance impact of converting every state value in the tree to a string on every render pass to put the string in this class directly. The idea is that if a particular collector of the debug stream needs to get the string representation en mass, it can do it on a background thread.